assets/js/modules/adsense/dashboard/adsense-module-status.js

```diff
@@ -52,43 +49,16 @@ class AdSenseModuleStatus extends Component {
 	}
 
 	async componentDidMount() {
-		this.iframeLoad();
-
-		await this.getExistingTag();
-
-		getAdSenseAccountStatus( this.updateLoadStatus, this.state.existingTag ).then( ( results ) => {
-			this.setState( results );
-		} );
-	}
-
-	/**
-	 * Save iframe for tag verification in state.
-	 */
-	iframeLoad() {
-		const iframe = isFrontendIframeLoaded();
-
-		if ( iframe && -1 !== iframe.dataset.modules.indexOf( 'adsense' ) ) {
-			const iFrameIsLoaded = iframe.contentDocument || ( iframe.contentWindow && iframe.contentWindow.document ) ;
-
-			if ( iFrameIsLoaded ) {
-				this.setState( { iframeLoaded: iframe } );
-			} else {
-				iframe.onload = async() => {
-					this.setState( { iframeLoaded: iframe } );
-				};
+		const existingTag = await getExistingTag( 'adsense' );
+
+		this.setState(
+			{ existingTag: 'no tag' === existingTag ? false : existingTag },
+			() => {
```

Comment:
@lucymtc I don't understand these lines of code.
While setting states, we running another function to set states?

Reply:
was setting a callback for setState as it's asynchronous and was using this.state.existingTag but there is no need to do this, I have changed it to be more readable in this commit https://github.com/google/site-kit-wp/pull/82/commits/765e0849a22219b38ddb1ada1f8a2590b44e5ce9